src/core/meshcop/tlvs.cpp

```diff
@@ -69,24 +69,28 @@ int Timestamp::Compare(const Timestamp &aCompare) const
     return rval;
 }
 
-ThreadError Tlv::GetTlv(const Message &message, Type type, uint16_t maxLength, Tlv &tlv)
+ThreadError Tlv::GetTlv(const Message &aMessage, Type aType, uint16_t aMaxLength, Tlv &aTlv)
 {
     ThreadError error = kThreadError_Parse;
-    uint16_t offset = message.GetOffset();
-    uint16_t end = message.GetLength();
+    uint16_t offset = aMessage.GetOffset();
+    uint16_t end = aMessage.GetLength();
 
     while (offset < end)
     {
-        message.Read(offset, sizeof(Tlv), &tlv);
+        Tlv tlv;
+        uint16_t totalLength;
+
+        aMessage.Read(offset, sizeof(tlv), &tlv);
+        totalLength = sizeof(tlv) + tlv.GetLength();
 
-        if (tlv.GetType() == type && (offset + sizeof(tlv) + tlv.GetLength()) <= end)
+        if (tlv.GetType() == aType && (offset + totalLength) <= end)
         {
-            if (maxLength > sizeof(tlv) + tlv.GetLength())
+            if (aMaxLength > totalLength)
```

Comment:
There are cases that intentionally only read out the front portion of the TLV (i.e. to process headers before digging further into the payload).